components/script/dom/bindings/buffer_source.rs

```diff
@@ -201,64 +201,85 @@ where
         }
     }
 
-    pub(crate) fn is_detached_buffer(&self) -> bool {
+    pub(crate) fn is_detached_buffer(&self, cx: JSContext) -> bool {
         match &self.buffer_source {
-            BufferSource::Int8Array(buffer)
-            | BufferSource::Int16Array(buffer)
-            | BufferSource::Int32Array(buffer)
-            | BufferSource::Uint8Array(buffer)
-            | BufferSource::Uint16Array(buffer)
-            | BufferSource::Uint32Array(buffer)
-            | BufferSource::Uint8ClampedArray(buffer)
-            | BufferSource::BigInt64Array(buffer)
-            | BufferSource::BigUint64Array(buffer)
-            | BufferSource::Float32Array(buffer)
-            | BufferSource::Float64Array(buffer)
-            | BufferSource::DataView(buffer)
-            | BufferSource::ArrayBuffer(buffer)
-            | BufferSource::Default(buffer) => unsafe {
+            BufferSource::Int8Array(buffer) |
+            BufferSource::Int16Array(buffer) |
+            BufferSource::Int32Array(buffer) |
+            BufferSource::Uint8Array(buffer) |
+            BufferSource::Uint16Array(buffer) |
+            BufferSource::Uint32Array(buffer) |
+            BufferSource::Uint8ClampedArray(buffer) |
+            BufferSource::BigInt64Array(buffer) |
+            BufferSource::BigUint64Array(buffer) |
+            BufferSource::Float32Array(buffer) |
+            BufferSource::Float64Array(buffer) |
+            BufferSource::DataView(buffer) |
+            BufferSource::Default(buffer) => {
+                assert!(self.is_initialized());
+                let mut is_shared = false;
+                unsafe {
+                    if JS_IsArrayBufferViewObject(*buffer.handle()) {
```

Comment:
Here we should assert, because these should all be views. Or just remove the condition like is done below. either way it should be consistent.